packages/itwin/tree-widget/src/components/TreeSelector.tsx

```diff
@@ -0,0 +1,52 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+
+import "./TreeSelector.scss";
+import type { SelectOption } from "@itwin/itwinui-react";
+import { useMemo, useState } from "react";
+import { Select } from "@itwin/itwinui-react";
+
+export interface TreeContentDefinition {
+  id: string;
+  label: string;
+  render: (density?: "enlarged" | "default") => React.ReactNode;
+}
+
+export interface TreeSelectorProps {
+  defaultSelectedContentId: string;
+  children: TreeContentDefinition[];
```

Comment:
I suggest renaming this to `definitions` or `trees` because it might be confusing as `children` prop name is used for child components.